include/shark/Algorithms/DirectSearch/Operators/Hypervolume/HypervolumeSubsetSelection2D.h

```diff
@@ -0,0 +1,298 @@
+/*!
+ *
+ * \author      O.Krause, T. Glasmachers
+ * \date        2014-2016
+ *
+ *
+ * \par Copyright 1995-2016 Shark Development Team
+ * 
+ * <BR><HR>
+ * This file is part of Shark.
+ * <http://image.diku.dk/shark/>
+ * 
+ * Shark is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published 
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ * 
+ * Shark is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ * 
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Shark.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+#ifndef SHARK_ALGORITHMS_DIRECTSEARCH_HYPERVOLUMESUBSETSELECTION_2D_H
+#define SHARK_ALGORITHMS_DIRECTSEARCH_HYPERVOLUMESUBSETSELECTION_2D_H
+
+#include <shark/LinAlg/Base.h>
+
+#include <algorithm>
+#include <vector>
+#include <deque>
+
+namespace shark {
+/// \brief Implementation of the exact hypervolume subset selection algorithm in 2 dimensions.
+///
+/// This algorithm solves the problem of selecting a subset of points with largest hypervolume in 2D.
+/// The algorithm has complexity n (k+log(n)) 
+///
+/// The algorithm is described in:
+/// Bringmann, Karl, Tobias Friedrich, and Patrick Klitzke. 
+/// "Two-dimensional subset selection for hypervolume and epsilon-indicator."
+/// Proceedings of the 2014 conference on Genetic and evolutionary computation. 
+/// ACM, 2014.	
+/// (although it is not very helpful)
+struct HypervolumeSubsetSelection2D {
+private:
+	
+	struct Point{
+		Point(){}
+
+		Point(double f1, double f2, std::size_t index)
+		: f1(f1)
+		, f2(f2)
+		, index(index)
+		, selected(false)
+		{}
+			
+		bool operator<(Point const& rhs) const{//for lexicographic sorting
+			if (f1 < rhs.f1) return true;
+			if (f1 > rhs.f1) return false;
+			return (f2 < rhs.f1);
+		}
+		
+		double f1;
+		double f2;
+		std::size_t index;
+		bool selected;
+	};
+	
+	///\brief Linear function a*x+b where a is stored in first and b is stored in section.
```

Comment:
The documentation for this type is not very helpful. What's the purpose of the type? Well, compute a linear function. Please provide a bit more context.